### PR TITLE
add metadata unit tests and markdown linting

### PR DESCRIPTION
**Pull request by christian-dinh**
### Description

- Added unit tests for validating metadata keys and value types
- Added `markdownlint` for validating markdown file content
- Added markdown files to `format` script and ran it

most of the changes are autofixes
- `"` -> `'`
- deleted excess whitespace and trailing spaces
- removed `$` from shell codeblocks that don't display output
- aligned table cells
- max line length = 120 chars

some changes I made by hand
- renamed a few duplicate `example` titles to `example 1`, `example 2`, etc
- deleted `Codecademy Hub Page` metadata keys from files that set it to `null`, updated docs to just omit it instead
- fixed a few typos

### Type of Change

- [ ] Adding a new entry
- [x] Editing an existing entry (fixing a typo, bug, issues, etc)
- [x] Updating the documentation

### Checklist

- [x] My entry follows the Codecademy Docs style guide
- [x] I have performed a self-review of my own writing and code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have checked my entry and corrected any misspellings
- [ ] All writings are my own

### Codecademy Username

If you want your Codecademy username be displayed in your entry (feature coming in October), make sure to have your Codecademy user profile linked to your GitHub:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the GitHub field, add your GitHub profile URL.

### CODE REVIEW

File: content/css/concepts/display/terms/clear/clear.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
 ---
-Title: "clear"
-Description: "Species whether an element coming after a floated element should be moved down or not."
+Title: 'clear'
+Description: 'Species whether an element coming after a floated element should be moved down or not.'
```
Comment (aionate0812) [inline content/css/concepts/display/terms/clear/clear.md:3]: Specifies?

Not related to the changes but maybe worth fixing the typo

File: content.test.ts [hunk 2]
```diff
@@ -49,3 +75,68 @@ describe('Codecademy Docs Content', () => {
...
+  it('has valid metadata values', () => {
+    const testOptionalString = (key: string, val?: string) => {
+      if (val !== undefined && typeof val !== 'string') {
+        expect(
+          `Expected ${key} to be a string or undefined but got ${typeof val}`
+        ).toBe('');
+      }
+    };
+
+    const testOptionalStringArray = (key: string, val?: string[]) => {
```
Comment (aionate0812) [inline content.test.ts:120]: TIL a way to test optional values

**Review by aionate0812 — APPROVED** [review]
LGTM, I like the way we are testing for optional values.

so many files the page crashed several times lol

File: content/css/concepts/display/terms/clear/clear.md (hunk 1 above)
Comment (christian-dinh) [reply content/css/concepts/display/terms/clear/clear.md:3]: hehe yea good catch

**Pull request merged**